### PR TITLE
DOCAPI-6425: Security settings

### PR DESCRIPTION
**Pull request by BayoNet**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Category (leave one):
- Documentation for #4792.

### CODE REVIEW

File: docs/en/operations/settings/settings_users.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
+# User settings
+
+The `users` section of the `user.xml` configuration file aggregates settings for users.
```
Comment (blinkov) [inline docs/en/operations/settings/settings_users.md]: "aggregates" is not really a fitting verb as common pattern is to put each user into separate file, each having <users> section

File: docs/en/operations/settings/settings_users.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    </user_name>
+    <!-- Other users settings -->
+</users>
+```
+
+### user_name/password
+
+Password could be specified in plaintext or in SHA256 (hex format).
+
+- To specify password in plaintext (not recommended), place it in a `password` element.
```
Comment (blinkov) [inline docs/en/operations/settings/settings_users.md]: I'd highlight this "not recommended" at least in bold or maybe red

File: docs/en/operations/settings/settings_users.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+    For example, `<password>qwerty</password>`. Password can be empty.
+
+- To specify SHA256 hash of a password, place it in a `password_sha256_hex` element.
+
+    For example, `<password_sha256_hex>65e84be33532fb784c48129675f9eff3a682b27168c0ea744b2cf58ee02337c5</password_sha256_hex>`.
+
+    To generate password, in the command prompt perform the command:
```
Comment (blinkov) [inline docs/en/operations/settings/settings_users.md]: Example of how to generate password from shell

File: docs/en/operations/settings/settings_users.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    The first line of the result is the password. The second line is the corresponding SHA256 hash.
+
+
+### user_name/networks
+
+List of networks which the user can connect to ClickHouse server from.
+
+Each element of list has one of the following forms:
+
+- `<ip>` — IP-address or a network mask.
```
Comment (blinkov) [inline docs/en/operations/settings/settings_users.md]: we could explicitly say that either IPv4 and IPv6 are supported, it's still not that common

File: docs/en/operations/settings/settings_users.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+    Examples: `213.180.204.3`, `10.0.0.1/8`, `10.0.0.1/255.255.255.0`, `2a02:6b8::3`, `2a02:6b8::3/64`, `2a02:6b8::3/ffff:ffff:ffff:ffff::`.
+
+- `<host>` — Hostname.
+
+    Example: `server01.yandex.ru`.
+
+    To check access, DNS query is performed, and all received addresses compared to peer address.
```
Comment (blinkov) [inline docs/en/operations/settings/settings_users.md]: * **are** compared
* received -> returned
* addresses -> IP-addresses

File: docs/en/operations/settings/settings_users.md [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+- `<host_regexp>` — Regular expression for host names.
```
Comment (blinkov) [inline docs/en/operations/settings/settings_users.md]: we'd better write hostname always in one word (or always in two)

File: docs/en/operations/settings/settings_users.md [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+
+    Example, `^server\d\d-\d\d-\d\.yandex\.ru$`
+
+    To check access, DNS PTR query is performed for peer address and then regexp is applied. Then, for result of PTR query, another DNS query is performed and all received addresses compared to peer address. Strongly recommended that regexp is ends with $
```
Comment (blinkov) [inline docs/en/operations/settings/settings_users.md]: PTR query could link to https://en.wikipedia.org/wiki/Reverse_DNS_lookup

File: docs/en/operations/settings/settings_users.md [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+
+All results of DNS requests are cached till server restart.
```
Comment (blinkov) [inline docs/en/operations/settings/settings_users.md]: here should be a link to a query that drops DNS cache

File: docs/en/operations/settings/settings_users.md [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+
+**Examples**
+
+To open access for user from any network, specify:
```
Comment (blinkov) [inline docs/en/operations/settings/settings_users.md]: should be accompanied by a warning that says that it's dangerous unless you have a firewall properly configured or server is not directly connected to Internet

File: docs/en/operations/settings/settings_users.md [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+To open access only from localhost, specify:
+
+```
+<ip>::1</ip>
+<ip>127.0.0.1</ip>
+```
+
+### user_name/profile
+
+You can assign a settings profile for the user. Settings profiles are configured in a separate section of the `users.xml` file. For more information see the [Settings profile](settings_profiles.md).
```
Comment (blinkov) [inline docs/en/operations/settings/settings_users.md]: "Settings profile**s**", but it still sounds weird
maybe "Profile settings"?

File: docs/en/operations/settings/settings_users.md [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+### user_name/quota
+
+Quotas allow you to limit resource usage over a period of time, or track the use of resources. Quotas are configured in the `quotas`
+section of the `users.xml` configuration file.
+
+You can assign a quotas set for the user. For the detailed description of quotas configuration see the [Quotas](../quotas.md#quotas) section.
+
+### user_name/databases
+
+In this section you can choose which rows ClickHouse returns by `SELECT` query. This feature implements a kind of Row-Level Security in ClickHouse.
```
Comment (blinkov) [inline docs/en/operations/settings/settings_users.md]: ...you can limit rows that are returned by ClickHouse for SELECT queries of current user, thus implementing basic row level security.

File: docs/en/operations/settings/settings_users.md [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+        <database_name>
+            <table1>
+                <filter>id = 1000</filter>
+            </table1>
+        </database_name>
+    </databases>
+</user1>
+```
+
+The `filter` can be any expression resulting with the [UInt8](../../data_types/int_uint.md)-typed value. It usually contains comparisons and logical operators. The filtering is incompatible with `PREWHERE` operations and disables `WHERE→PREWHERE` optimization.
```
Comment (blinkov) [inline docs/en/operations/settings/settings_users.md]: should explicitly say that rows from `database_name.table1` where `filter` results to 0 are not returned for this user

File: docs/toc_en.yml [hunk 13]
```diff
@@ -183,6 +183,7 @@ nav:
     - 'Restrictions on Query Complexity': 'operations/settings/query_complexity.md'
     - 'Settings': 'operations/settings/settings.md'
     - 'Settings Profiles': 'operations/settings/settings_profiles.md'
+    - 'User settings': 'operations/settings/settings_users.md'
```
Comment (blinkov) [inline docs/toc_en.yml]: **S**ettings

**Review by blinkov — CHANGES_REQUESTED** [review]

File: docs/en/operations/settings/settings_users.md (hunk 4 above)
Comment (BayoNet) [reply docs/en/operations/settings/settings_users.md]: There is an example that shows it. Or I understood your comment wrong?

File: docs/en/operations/settings/settings_users.md (hunk 8 above)
Comment (BayoNet) [reply docs/en/operations/settings/settings_users.md]: It is not described. I'll create a ticket.

**Pull request merged**